QUARKUS-2727.md

```diff
@@ -0,0 +1,98 @@
+- using multiple secret keys handlers:
+  - custom secret keys handlers
+  - Jasypt secret keys handler
+  - out-of-the-box handlers provided by Crypto (`smallrye-config-crypto` dependency)
+- modification of secret keys expressions and secret keys handlers in DEV mode
+- secret keys handlers in native mode and make sure all the handlers (custom and built in) are registered for reflection automatically
+- secrets hidden from accidental exposure by `SecretKeysConfigSourceInterceptor`, that is secrets that are also Secret Keys
+- encryption key used to decode secrets encoded by the `AES/GCM/NoPadding` algorithm. The point here is to verify algorithm emphasized in upstream documentation.
+- basic encryption capabilities provided by Jasypt handler
+
+### Existing tests
+
+The Quarkus Test Suite contains Secret Keys Names in its config module.
+Customers are likely to use them together with secrets and so should we.
+Upstream coverage is mostly secured by SmallRye Config project and test coverage inside Quarkus is sporadic.
+That means we need fully verify functionality in DEV mode and native mode.
+
+## Impact on test suites and test environment
+
+New scenarios are going to be added to the `config` module of the Quarkus Test Suite.
+Integrating scenarios into existing tests `ConfigIT`, `OpenShiftFileSystemConfigSecretConfigIT` and `OpenShiftApiServerConfigSecretConfigIT`
+will not only be considerate to the resources (we don't need to build and start new application), but also makes sense content wise.
+We already test there [Secret Keys Names](https://smallrye.io/smallrye-config/Main/config/secret-keys/#secret-keys-names)
+and OpenShift secrets respectively.
+
+New DEV mode test will be required to make sure changes in watched files are reflected and secrets updated.
+We should also update secret keys expressions (exchange handlers and so on).
+
+### Impact on resources:
+
+- No additional requirements for resources in lab
+- Required additional time for the test execution should be around 1 minute for tests added to existing classes (please see above).
+  That includes HTTP requests, secrets loading, encryption and decryption and result evaluation. 
+  The DEV mode test will add extra 1 minute to test execution time.
+- No measurable additional requirements for a Native image scenario
+
+## Getting familiar with the feature
+
+Following actions were taken to ensure familiarity:
+- Ensure documentation provides clear explanation on configuration options
+- Ensure good user experience and simplicity of use
+
+## Advanced topics for test development
+
+- Apply additional converters, config sources and interceptors on secrets
+
+## Future considerations
+
+- SmallRye Config [plans following improvements](https://github.com/smallrye/smallrye-config/pull/833):
+  - ability to configure SecretKeysHandler
+  - certificate support
+  - provide out of the box handlers
+- Quarkus CLI might have a plugin, that will help with encrypting values
```

Comment:
The plugin is imho part of QUARKUS-2727, maybe worth clarifying when it comes. Pls ping Max on Zulip.

Reply:
Yes, this is intentionally vague as I don't know if this information is pubilc @rsvoboda

Reply:
But I'll check with him.